corehq/motech/repeaters/tasks.py

```diff
@@ -30,34 +36,27 @@
 )
 def check_repeaters():
     start = datetime.utcnow()
-    cutoff = start + CHECK_REPEATERS_INTERVAL
+    repeater_lock_timeout = RECORD_LOCK_TIMEOUT // 2
+    timeout_dt = start + timedelta(seconds=repeater_lock_timeout)
 
-    # Timeout for slightly less than periodic check
+    # Long timeout to allow all waiting repeat records to be iterated
```

Comment:
Agreed. In fact this structure makes more sense regardless. In this case, does it make sense to have a timeout at all? Or is waiting a whole day to iterate all repeat records just ridiculous, and we should drop it to, say, 6 hours? (This guesswork drives me nuts. What about 24 hours, log how long it takes, and drop it to that + 50%?)

Reply:
> In this case, does it make sense to have a timeout at all?

I think timeouts are always used on redis locks in case of a failure that causes the lock to not be released (interpreter crash where `finally` clause is not executed, for example). Since the lock state is external to the process we want to make sure it is definitely released at some point.

However, I would argue that we should never rely on timeouts as part of the program happy path. Instead, the timeout should never be hit except for very rare and unexpected error scenarios. Ideally such errors should report themselves loudly so we know when they are occurring. If the frequency is high it is a sign that we need to re-think the design or at the very least the timeout.

> Or is waiting a whole day to iterate all repeat records just ridiculous, and we should drop it to, say, 6 hours? (This guesswork drives me nuts. What about 24 hours, log how long it takes, and drop it to that + 50%?)

Agree about the guesswork driving nuts. Might be useful to go to voice and try to work out a design that balances timeout length with implementation complexity. What do you think?